### PR TITLE
Enabled building GCC 13.1 on darwin/aarch64.

### PR DESCRIPTION
**Pull request by benc303**
This branch adds the capability to build gcc on Macs with Apple Silicon CPUs. This is useful for obtaining gfortran fully within Spack. It uses the patch from Homebrew to make this work; bootstrapping doesn't work so this is disabled using a conflict.

### CODE REVIEW

File: var/spack/repos/builtin/packages/gcc/package.py [hunk 1]
```diff
@@ -337,11 +337,11 @@ class Gcc(AutotoolsPackage, GNUMirrorPackage):
     #   on XCode 12.5
     conflicts("+bootstrap", when="@:11.1 %apple-clang@12.0.5")
 
-    # aarch64/M1 is supported in GCC 11.3-12.2
+    # aarch64/M1 is supported in GCC 11.3-12.2 and 13.1
     conflicts(
-        "@:11.2,12.3:",
+        "@:11.2,12.3",
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/gcc/package.py]: The conflict change seems wrong, as it does not represent the `11.3-12.2` + `13.1` sets of versions. Can we turn this conflict into a `requires`?

**Review by alalazo — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/gcc/package.py (hunk 1 above)
Comment (benc303) [reply var/spack/repos/builtin/packages/gcc/package.py]: Thanks for getting to this—the PR was submitted before 13.2 was added. I'll see if 13.2 can be made to work in a similar way, and switch this over to a `requires`.

Comment (benc303) [reply var/spack/repos/builtin/packages/gcc/package.py]: Done. Can the review be finished?

**Review by alalazo — APPROVED** [review]

**Comment by alalazo** [comment]
Screenshot from 2023-09-18 18-19-17

**Comment by alalazo** [comment]
Thanks!

**Pull request merged**